python/tests/catalog/test_rest.py

```diff
@@ -677,6 +677,120 @@ def test_create_table_409(rest_mock: Mocker, table_schema_simple: Schema) -> Non
     assert "Table already exists" in str(e.value)
 
 
+def test_register_table_200(rest_mock: Mocker, table_schema_simple: Schema) -> None:
```

Comment:
Test for the unhappy case? (409 in case the table already exists at the time of registration)

Reply:
Yes, I can add that one as well. For some context, in https://github.com/apache/iceberg/pull/8374 I've moved to integration tests, and I don't like mocking all of this, so I figured adding the other tests once that's implemented by the REST catalog.

Reply:
I've added te test :) Good suggestion, thanks!

Reply:
Thanks, this looks great!